src/shogun/regression/GLM.cpp

```diff
@@ -0,0 +1,100 @@
+#include <shogun/lib/config.h>
+
+#include <shogun/labels/RegressionLabels.h>
+#include <shogun/lib/SGVector.h>
+#include <shogun/mathematics/Math.h>
+#include <shogun/mathematics/linalg/LinalgNamespace.h>
+#include <shogun/mathematics/linalg/LinalgSpecialPurposes.h>
+
+#include <shogun/regression/GLM.h>
+#include <utility>
+using namespace shogun;
+
+GLM::GLM() : LinearMachine()
+{
+	init();
+}
+float64_t GLM::log_likelihood(
+    const std::shared_ptr<DenseFeatures<float64_t>>& features,
+    const std::shared_ptr<Labels>& label)
+{
+	auto vector_count = features->get_num_vectors();
+	auto feature_count = features->get_num_features();
+	ASSERT(vector_count > 0 && label->get_num_labels() == vector_count)
+
+	SGVector<float64_t> lambda(vector_count);
+	SGVector<float64_t> beta = LinearMachine::get_w();
+	float64_t beta0 = LinearMachine::get_bias();
```

Comment:
Do we override get_w/get_bias? As in is there a particular reason why you specify LinearMachine::?

Reply:
For some reason, I get get_w not declared in this scope without me doing so

Reply:
i'm not so sure how includes changed things...